crates/hir-ty/src/infer/coerce.rs

```diff
@@ -227,10 +227,16 @@ impl InferenceContext<'_> {
         expr: Option<ExprId>,
         from_ty: &Ty,
         to_ty: &Ty,
+        // [Comment from rustc](https://github.com/rust-lang/rust/blob/4cc494bbfe9911d24f3ee521f98d5c6bb7e3ffe8/compiler/rustc_hir_typeck/src/coercion.rs#L85-L89)
+        // Whether we allow `NeverToAny` coercions. This is unsound if we're
+        // coercing a place expression without it counting as a read in the MIR.
+        // This is a side-effect of HIR not really having a great distinction
+        // between places and values.
+        coerce_never: bool,
```

Comment:
Nit but we might want to make this a binary enum `CoerceNever::Yes/No` :^) so that its more clear to callers (we have an assist for that)

Reply:
That would be much less confusing 👍